core/src/test/java/org/opensearch/sql/expression/datetime/DateTimeFunctionTest.java

```diff
@@ -660,6 +660,47 @@ public void invalidDayOfYearArgument() {
             () ->  invalidDayOfYearQuery("asdfasdfasdf"))
     );
   }
+
+  private static Stream<Arguments> getTestDataForExtractFunction() {
+    return Stream.of(
+        Arguments.of("MICROSECOND", 123000),
+        Arguments.of("SECOND", 12),
+        Arguments.of("MINUTE", 11),
+        Arguments.of("HOUR", 10),
+        Arguments.of("DAY", 11),
+        Arguments.of("WEEK", 6),
+        Arguments.of("MONTH", 2),
+        Arguments.of("QUARTER", 1),
+        Arguments.of("YEAR", 2023),
+        Arguments.of("SECOND_MICROSECOND", 12123000),
+        Arguments.of("MINUTE_MICROSECOND", 1112123000),
+        Arguments.of("MINUTE_SECOND", 1112),
+        Arguments.of("HOUR_MICROSECOND", 101112123000L),
+        Arguments.of("HOUR_SECOND", 101112),
+        Arguments.of("HOUR_MINUTE", 1011),
+        Arguments.of("DAY_MICROSECOND", 11101112123000L),
+        Arguments.of("DAY_SECOND", 11101112),
+        Arguments.of("DAY_MINUTE", 111011),
+        Arguments.of("DAY_HOUR", 1110),
+        Arguments.of("YEAR_MONTH", 202302)
+    );
+  }
+
+
+  @ParameterizedTest(name = "{0}")
+  @MethodSource("getTestDataForExtractFunction")
+  public void testExtract(String part, long expected) {
+    lenient().when(nullRef.valueOf(env)).thenReturn(nullValue());
+    lenient().when(missingRef.valueOf(env)).thenReturn(missingValue());
+
+    FunctionExpression expression = DSL.extract(
+        DSL.literal(part),
+        DSL.literal(new ExprDatetimeValue("2023-02-11 10:11:12.123")));
```

Comment:
Test with other input types, especially with TIME and DATE -- extracting date/mixed parts from TIME and time/mixed parts from DATE

Reply:
Added in https://github.com/Bit-Quill/opensearch-project-sql/pull/221/commits/d15c26ba044d8db36dad2dcf197757404aeb8ece and 4dccd2358d4f372469933ef1d5deea9125b4e890. Still need to be modified according to other comments.